### ISSUE TITLE
Migrate from libmpv to libplacebo

### ISSUE THREAD

**Issue by Foul-Tarnished** [iina/iina#5203]
afaik, libmpv doesn't support Vulkan/MoltenVK, and is stuck with old OpenGL

Using LibPlacebo, we can get MoltenVK alongside proper HDR etc.

https://github.com/haasn/libplacebo

- High-quality, optimized upscaling and downscaling including support for polar filters ("Jinc"), anti-aliasing, anti-ringing and gamma correct scaling.
- Dynamic HDR tone mapping, including real-time measurement of scene histogram, scene change detection, dynamic exposure control, perceptual gamut stretching, contrast recovery and more.
- Native support for Dolby Vision HDR, including Profile 5 conversion to HDR/PQ or SDR, reading DV side data, and reshaping. (BL only, currently)
- A colorimetrically accurate color management engine with support for soft gamut mapping, ICC profiles, accurate ITU-R BT.1886 emulation, black point compensation, and custom 3DLUTs (.cube).
- A pluggable, extensible [custom shader system](http://libplacebo.org/custom-shaders/). This can be used to arbitrarily extend the range of custom shaders to include popular user shaders like RAVU, FSRCNNX, or Anime4K. See the [mpv wiki on user scripts](https://github.com/mpv-player/mpv/wiki/User-Scripts#user-shaders) for more information.
- High performance film grain synthesis for AV1 and H.274, allowing media players to offload this part of decoding from the CPU to the GPU.
- Tunable, fast debanding and deinterlacing shaders.
- High quality gamma-correct dithering, including error diffusion modes.

**Comment by Foul-Tarnished** [comment]
Until then, I guess mpv with gpu-next is superior to IINA

**Comment by low-batt** [comment]
Yes, there are many very desirable improvements in [GPU Next](https://github.com/mpv-player/mpv/wiki/GPU-Next-vs-GPU#general-differences) that are not currently available to clients using `libmpv`. 

At the moment IINA's plan is to wait for the mpv project to address issue https://github.com/mpv-player/mpv/issues/10810, "Support for libplacebo (gpu-next) in the rendering API".

**Comment by Foul-Tarnished** [comment]
> Yes, there are many very desirable improvements in [GPU Next](https://github.com/mpv-player/mpv/wiki/GPU-Next-vs-GPU#general-differences) that are not currently available to clients using `libmpv`.
> 
> At the moment IINA's plan is to wait for the mpv project to address issue [mpv-player/mpv#10810](https://github.com/mpv-player/mpv/issues/10810), "Support for libplacebo (gpu-next) in the rendering API".

This issue seems to ask for gpu-next features in their opengl backend inside libmpv though ?
Would be better to use macvk backend, that has gpu-next features, and is made for mac

**Comment by low-batt** [comment]
There was talk of a direct [Metal](https://developer.apple.com/metal/) implementation.

My sense is that the mpv project is still very focused on fixing `gpu-next` issues. For example mpv with [--target-colorspace-hint](https://mpv.io/manual/stable/#options-target-colorspace-hint) crashes a lot of the time while quitting. So I've been keeping quiet and not pestering the mpv project on how they plan on exposing the new features to `libmpv` clients.

IINA has fallen behind and is currently focused on upgrading dependencies, adding support for new mpv features such as the additional controls on secondary subtitles and addressing some long standing technical debt. We are getting close to a beta release that contains these changes.
